src/useTransition.js

```diff
@@ -32,7 +32,7 @@ const makeConfig = props => {
 
 export function useTransition(input, keyTransform, props) {
   // Coerce props to an object
-  props = callProp(props)
+  props = useMemo(() => callProp(props), [])
```

Comment:
Nope, still not good enough. :)

This ends up caching `props` when it's an object, which isn't what we want.

Reply:
Or is it? :P

Reply:
for transition im not sure tbh, it's undefined atm. i never considered it updateable, items can update, but i've never seen someone mutate the config (it also doesn't have an updater for this reason).

Reply:
For all hooks, re-rendering should always queue an update (unless a deps array is given). Just for consistency.

Reply:
for the "update" lifecycle perhaps, yes. But the others are one-time slots. Enter only happens once, or leave, there's there that can be updated. Update is a valid concern though, so how do we want to define it? And is this something that v9 useTransition can do already?

Reply:
The `enter` and `leave` props should still be updatable for future enters and leaves.

PS: We may want to remove the `update` prop now that the `update` function is always returned.

Reply:
Makes sense (enter/leave). The update prop is useful, it comes from d3 and has a specific meaning. It fires for all nodes that aren't either enter or leave. Like so: https://codesandbox.io/embed/26mjowzpr

Reply:
It might be cool to allow custom phases instead of (or in addition to?) the "update" phase. The custom phase would be derived from each item. I haven't thought much about it yet, but it *could* be useful for multi-phase transitions.